### DISCUSSION TITLE
Создание интентов через триггер conversation, как идея на будущее

### DISCUSSION THREAD

**Discussion by malinovsku** [dext0r/ha-yandex-station-intents#43] [Ideas]
Ранее в ТГ https://t.me/yandex_smart_home/77603 было предложение по доработке компонента, хотел бы написать что я нашел и что в итоге получил)
Суть идеи была добавлять триггеры для интентов через platform: conversation в авто или скрипте, при создании нового триггера как то получать все "предложения триггеров" (полазил по исходникам, нашел что оно хранится и проверяется, но не понял можно и как получить их https://github.com/home-assistant/core/blob/858fb1fa376af47e570b0a9f718502de2b0d2636/homeassistant/components/conversation/default_agent.py#L153) и добавлять их в список поштучно как это сейчас работает через ямл. 
Это что касается получения списка не из ямла, но вот то что касается просто вызова триггера conversation, то там минимум добавить вот тут после https://github.com/dext0r/ha-yandex-station-intents/blob/0ede79c744d237702a67913b35df17fd6a95d912/custom_components/yandex_station_intents/yandex_intent.py#L127
Что то типа такого, у меня сработало, скрин.
```
from homeassistant.components import conversation

result = await conversation.async_converse(
                hass=self._hass,
                text=event_data['text'],
                language='ru',
                conversation_id=event_data['entity_id'],
                context=None,
            )
```
![Снимок экрана 2024-01-27 в 21 15 52](https://github.com/dext0r/ha-yandex-station-intents/assets/34610364/4f83db6e-5a0d-45f7-b9d7-f0f8a0efa216)

и там так же на подходе https://github.com/home-assistant/core/pull/108231  , где будет возможность из авто вернуть ответ в  conversation, аналог say_phrase, сейчас на все ответ Done.
Теперь о плохом :) триггер conversation не несет никакой инфы кроме текста или объектов в тексте и дополнительно добавить что то нельзя https://github.com/home-assistant/core/blob/858fb1fa376af47e570b0a9f718502de2b0d2636/homeassistant/components/conversation/trigger.py#L79 

Все это на обсудить, компонент работает безупречно и сейчас, но цель всей этой суеты - нативность таких простых вещей, небыло бы необходимости залазить в ямл, прописывать, потом опять прописывать в триггере событие, а только от силы может перезагрузить компонент. Если идея пойдет, то можно было и в ядре попросить какую то полезную инфу еще в триггере этом.
Так же вся эта идея, в полном воплощении без ямла, делает бессмысленной текущую логику компонента с say_phrase execute_command , дополнительными фразами, так как для этого по сути есть все средства в ха без хардкорства и все на виду.

**Comment by malinovsku** [comment]
Дополню по шагам без лишнего
  1. пользователь добавляет фразы через триггер/ы Conversation, это и авто и ожидание в скриптах
  2. Компонент прослушивает события изменения объектов автоматизаций и скриптов entity_registry_updated, после получает как то список всех фраз триггеров Conversation и сверяет уникальные фразы с внутренним списком уже добавленных фраз, если есть расхождения, то выполняет обновление/добавление/удаление сценариев в Яндексе. Либо пользователь сам ребутит компонент после всех манипуляций с автоматизацией и компонент так же находит все триггеры, сверяет добавляет/удаляет..
  3. при активации добавленной фразы на колонке - запускается триггер platform: conversation с добавленной фразой

*Фишка с альтернативными фразами для фраз может создать двоечтение, как предложение 1 фраза - 1 сценарий - множество триггеров с этой фразой в ха, без альтернатив, при необходимости добавить еще фраз можно перечислением в триггере и они никак не связаны будут между собой в Яндексе, каждый отдельно.

**Comment by dext0r** [comment]
Идея интересная, но есть большая проблема. Цитата из [документации](https://www.home-assistant.io/docs/automation/trigger/#sentence-trigger) к триггеру по conversation:
> Sentence triggers only work with Home Assistant Assist. External conversation agents such as OpenAI or Google Generative AI cannot be used to trigger automations.


Ну или я не полностью понял задумку.

**Comment by malinovsku** [comment]
> Идея интересная, но есть большая проблема. Цитата из [документации](https://www.home-assistant.io/docs/automation/trigger/#sentence-trigger) к триггеру по conversation:
> 
> > Sentence triggers only work with Home Assistant Assist. External conversation agents such as OpenAI or Google Generative AI cannot be used to trigger automations.
> 
> Ну или я не полностью понял задумку.

Я понимаю эту вырезку как то что условно сгенерированное ИИ не может быть триггером, а только то что сказали голосом assistу, по сути тоже самое и у Алисы(опуская детали интеграции)
